File: WebApp.Core/Entities/Compensation/PaymentTableHierarchy.cs
namespace WebApp.Core.Entities.Compensation
{
    using WebApp.Core.Entities.Hierarchy;

    public class PaymentTableHierarchy : IEntityBase
    {
        public int PaymentTableId { get; set; }

        public HierarchyStructure HierarchyStructure { get; set; }

        public int HierarchyStructureId { get; set; }

        public HierarchyStructureNode HierarchyStructureNode { get; set; }

        public int HierarchyStructureNodeId { get; set; }

        public int Id { get; set; }
    }
}

File: BaseWebApp/ClientApp/src/app/core/models/compensation/PaymentTableHierarchy.ts
import { IHierarchyStructure, IHierarchyStructureNode } from "../hierarchy";

export interface IPaymentTableHierarcy {
    hierarchyStructure: IHierarchyStructure;
    hierarchyStructureId: number;
    hierarchyStructureNode: IHierarchyStructureNode;
    hierarchyStructureNodeId: number;
}